crates/relayer/src/chain/ckb4ibc.rs

```diff
@@ -1483,39 +1507,176 @@ impl ChainEndpoint for Ckb4IbcChain {
+            .remove(&commitment_path);
+
+        Ok((Some(client_state), proof))
     }
 
     fn build_channel_proofs(
         &self,
-        _port_id: &PortId,
-        _channel_id: &ChannelId,
+        port_id: &PortId,
+        channel_id: &ChannelId,
         height: Height,
     ) -> Result<Proofs, Error> {
-        // FIXME: fix it until the hardfork of CKB
-        Ok(get_dummy_merkle_proof(height))
+        // use dummy merkle proof when the counterparty is aslo CKB
+        if matches!(self.counterparty_client_type(), ClientType::Ckb4Ibc) {
+            return get_ibc_merkle_proof(height, vec![0u8]);
+        }
+
+        let commitment_path = channel_path(port_id.as_str(), channel_id.as_str());
+        let mut tx_hash = self
+            .ibc_transactions_cache
+            .lock()
+            .unwrap()
+            .get(&commitment_path)
+            .cloned();
+
+        // search frist channel cell on-chain if cache is missing
+        if tx_hash.is_none() {
+            let channel_key = get_channel_search_key(
+                &self.config,
+                self.counterparty_client_type(),
+                channel_id,
+                port_id,
+                true,
+            )?;
+            let result =
+                self.rt
+                    .block_on(self.rpc_client.fetch_live_cells(channel_key, 1, None))?;
+            let Some(cell) = result.objects.first() else {
+                return Err(Error::other_error(format!(
+                    "no channel event found for {channel_id}/{port_id} at block {height}"
+                )));
+            };
+            tx_hash = Some(cell.clone().out_point.tx_hash);
+        }
+
+        let tx_hash = tx_hash.unwrap();
+        let Some(proof) = self.rt.block_on(generate_tx_proof_from_block(
+            self.rpc_client.as_ref(),
+            &tx_hash,
+        ))?
+        else {
+            return Err(Error::other_error(format!(
+                "cannot generate channel proof for tx {} at block {height}",
+                hex::encode(tx_hash)
+            )));
+        };
+
+        self.ibc_transactions_cache
+            .lock()
+            .unwrap()
+            .remove(&commitment_path);
+
+        Ok(proof)
     }
 
     fn build_packet_proofs(
         &self,
         _packet_type: PacketMsgType,
-        _port_id: PortId,
-        _channel_id: ChannelId,
-        _sequence: Sequence,
+        port_id: PortId,
+        channel_id: ChannelId,
+        sequence: Sequence,
         height: Height,
     ) -> Result<Proofs, Error> {
-        // FIXME: fix it until the hardfork of CKB
-        Ok(get_dummy_merkle_proof(height))
+        // use dummy merkle proof when the counterparty is aslo CKB
+        if matches!(self.counterparty_client_type(), ClientType::Ckb4Ibc) {
+            return get_ibc_merkle_proof(height, vec![0u8]);
+        }
+
+        let commitment_path =
+            packet_commitment_path(port_id.as_str(), channel_id.as_str(), sequence.into());
```

Comment:
There are two different kinds of packet proofs: packet and ack (see the `packet_type` argument). How does this handle ack commitment?

Reply:
I found packet type is passed by this method, I can use this argument to distinguish

Reply:
Are they distinguished when stored to the cache?

Reply:
no, this should be fixed

Reply:
check again